### PR TITLE
Add separate `register` route

### LINKED ISSUE #5076: Separate login and signup pages
In order to integrate the new landing page for webknossos.org, we need separate pages for login and signup.
Preferably, the url would be `/login` and `/signup`.

### PR DESCRIPTION
**Pull request by philippotto**
### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- (test with isDemo true and false separately)
- open `/register` and create new account

### Issues:
- fixes #5076

------
- [x] Updated [(unreleased) changelog](../blob/master/CHANGELOG.unreleased.md#unreleased)
- [ ] Updated [(unreleased) migration guide](../blob/master/MIGRATIONS.unreleased.md#unreleased) if applicable
- [ ] Updated [documentation](../blob/master/docs) if applicable
- [ ] Adapted [wk-connect](https://github.com/scalableminds/webknossos-connect) if datastore API changes
- [ ] Needs datastore update after deployment
- [x] Ready for review

### CODE REVIEW

**Comment by philippotto** [comment]
Originally, I also planned to refactor the authentication forms a bit, but gave up in the end, since it got out of hand. Instead, I created a follow-up issue which hopefully captures the insights I gained today: https://github.com/scalableminds/webknossos/issues/5096

File: frontend/javascripts/router.js [hunk 1]
```diff
@@ -427,10 +427,7 @@ class ReactRouter extends React.Component<Props> {
               />
               <Route
                 path="/auth/register"
```
Comment (normanrz) [inline frontend/javascripts/router.js:429]: Would you mind renaming the route to `/auth/signup`?

Comment (philippotto) [reply frontend/javascripts/router.js:429]: Should we keep the old `register` path (and redirect to `signup`)? Or can it be a "breaking change"?

Comment (normanrz) [reply frontend/javascripts/router.js:429]: should be a redirect or keep both.

Comment (philippotto) [reply frontend/javascripts/router.js:429]: done

File: frontend/javascripts/admin/auth/registration_view.js [hunk 2]
```diff
@@ -4,15 +4,12 @@ import { Link, useHistory } from "react-router-dom";
...
-import RegistrationForm from "./registration_form";
+import SpotlightRegistrationForm from "admin/auth/spotlight_registration_form";
+import RegistrationForm from "admin/auth/registration_form";
 
-type Props = {
-  organizationName?: string,
-};
-
-function RegistrationView({ organizationName }: Props) {
+function RegistrationViewNotDemo() {
```
Comment (daniel-wer) [inline frontend/javascripts/admin/auth/registration_view.js:12]: I was confused how invite links could still work, although no organization name could be supplied, here, but I found out that for invites the AuthenticationModal is used. Glad you created https://github.com/scalableminds/webknossos/issues/5096 :+1:

**Review by daniel-wer — APPROVED** [review]
I tested the demo, non-demo and invitation registration case. Everything worked as expected, nice work :+1: Would definitely be in favor of https://github.com/scalableminds/webknossos/issues/5096 :)

**Pull request merged**